src/main.cpp

```diff
@@ -5,66 +5,68 @@
 
 using std::cout;
 using std::endl;
+using std::cerr;
 using glm::vec3;
 
-int main()
-{
-	// {
-	// 	glm::quat q(glm::radians(vec3(30.0f, 40.0f, 50.0f)));
-	// 	vec3 v(glm::degrees(eulerAngles(q)));
-	// 	cout << q.w << " " << q.x << " " << q.y << " " << q.z << endl;
-	// 	cout << v.x << " " << v.y << " " << v.z << endl;
-	// 	cout << endl;
-	// }
-
-	// {
-	// 	GTAQuat q(vec3(30.0f, 40.0f, 50.0f));
-	// 	vec3 v = q.ToEuler();
-	// 	cout << q.w << " " << q.x << " " << q.y << " " << q.z << endl;
-	// 	cout << v.x << " " << v.y << " " << v.z << endl;
-	// 	cout << endl;
-	// }
+/*
+	Epsilon types:
+		0 - zero (default)
+		1 - ieee
+		2 - illidan
 
-	// {
-	// 	GTAQuat q(-0.237544f, -0.237661f, 0.665970f, -0.666012f);
-	// 	vec3 v = q.ToEuler();
-	// 	cout << q.w << " " << q.x << " " << q.y << " " << q.z << endl;
-	// 	cout << v.x << " " << v.y << " " << v.z << endl;
-	// 	cout << endl;
-	// }
+	Terminology:
+		A = original quat
+		B = quat_to_euler(A)
+		C = euler_to_quat(B)
+		D = quat_to_euler(C)
 
-	// {
-	// 	GTAQuat q(-0.237602f, -0.237603f, 0.665991f, -0.665991f);
-	// 	vec3 v = q.ToEuler();
-	// 	cout << q.w << " " << q.x << " " << q.y << " " << q.z << endl;
-	// 	cout << v.x << " " << v.y << " " << v.z << endl;
-	// 	cout << endl;
-	// }
+	Output:
+		1. original quat A
+		2. epsilon value used
+		3. space separated differences between A<->C, B<->D
+*/
 
-	// {
-	// 	GTAQuat q(-0.237544f, -0.237661f, 0.665970f, -0.666012f);
-	// 	vec3 v = q.ToEuler();
-	// 	cout << q.w << " " << q.x << " " << q.y << " " << q.z << endl;
-	// 	cout << v.x << " " << v.y << " " << v.z << endl;
+int main(int argc, char **argv)
+{
+	
+	if (argc != 6) 
+	{
+		cerr << "Usage: /<name> <qw> <qx> <qy> <qz> <epsilon_type>" << endl;
+		exit(1);
+	}
 
-	// 	GTAQuat q2(v);
-	// 	cout << "==> " << q2.w << " " << q2.x << " " << q2.y << " " << q2.z << endl;
-	// 	vec3 v2 = q2.ToEuler();
-	// 	cout << "==> " << v2.x << " " << v2.y << " " << v2.z << endl;
-	// 	cout << endl;
-	// }
+	float qw = (float)atof(argv[1]);
+	float qx = (float)atof(argv[2]);
+	float qy = (float)atof(argv[3]);
+	float qz = (float)atof(argv[4]);
+	short epsilon = atoi(argv[5]);
 
+	if(epsilon == 1) 

```

Comment:
Could've used switch no?

Reply:
Yes, but since there's only 3 possibilities it felt more suitable to use just if-else if-else structure over switch. To be honest, I don't think it'd be that much of a difference as the cases would be contiguous so it could be implemented as a switch statement too.